src/analyzing/Dataflow_svalue.ml

```diff
@@ -406,18 +406,32 @@ let no_cycles_in_svalue (id_info : G.id_info) svalue =
      * we can have a single visitor for all calls, given that the old
      * `mk_visitor` was pretty expensive, and constructing a visitor object may
      * be as well. *)
+    let i = ref 0 in
     let ff = ref (fun _ -> assert false) in
     let ok = ref true in
     let vout =
-      object
-        inherit [_] G.iter
+      object (self : 'self)
+        inherit [_] G.iter_no_id_info
```

Comment:
Why do we `iter_no_id_info` when we're visiting `id_info`? Curious for my understanding

Reply:
We're overwriting the `visit_id_info` and not using the one from `super` so we can use any of them really. But I thought it was slightly more clear to inherit from `iter_no_id_info` because it does not visit `id_info` so you immediately know that `visit_id_info` does not depend on inherited behavior.